axelor-sale/src/main/java/com/axelor/apps/sale/service/SaleOrderGeneratorServiceImpl.java

```diff
@@ -94,4 +97,16 @@ protected void checkClientPartner(Partner clientPartner, SaleOrder saleOrder)
           company.getName());
     }
   }
+
+  protected void checkinAti(SaleOrder saleOrder, String inAti) throws AxelorException {
+    Boolean ati = Boolean.parseBoolean(inAti);
+    Company company = saleOrder.getCompany();
+    if (company.getSaleConfig().getSaleOrderInAtiSelect() == 1
```

Comment:
To fetch the company sale config, do not get it directly from the company. Use SaleConfigService#getSaleConfig. You should also check that inAti is not null since you should change it to a Boolean.

Instead of using hard values, use values from SaleConfigRepository.